bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -264,6 +270,12 @@ fn main() -> Result<()> {
                             debug!("Metric event: {:?}", event);
                         }
                         Event::Telemetry(event) => match event.record {
```

Comment:
Main function is growing a lot with different abstraction levels and loops. Is it reasonable to extract a few methods to make more readable?

Reply:
Yeah, I think it'd make sense to start figuring out how to split them – let's log it!